src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,116 @@
+package duke.ui;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.DukeException;
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnMarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * A class that parses the input from user into something program can understand.
+ */
+public class Parser {
+
+    private enum Inputs {
+        BYE,
+        LIST,
+        MARK,
+        UNMARK,
+        TODO,
+        DEADLINE,
+        EVENT,
+        DELETE,
+        FIND,
+        ELSE
+    }
+
+    /**
+     * Returns a Command after parsing through user's String command.
+     *
+     * @param command A String command that the user inputs.
+     * @return The Command that user wants to execute.
+     * @throws DukeException If there is an invalid command.
+     * @throws DateTimeParseException If there is an error in Date and Time input.
+     */
+    public static Command parse(String command) throws DukeException, DateTimeParseException {
+        switch (getInput(command)) {
+        case BYE:
+            return new ByeCommand();
+            // Fallthrough
+        case LIST:
+            return new ListCommand();
+            // Fallthrough
+        case MARK:
+            int markNum = Integer.parseInt(command.replace("mark ", ""));
+            return new MarkCommand(markNum - 1);
+            // Fallthrough
+        case UNMARK:
+            int unMarkNum = Integer.parseInt(command.replace("unmark ", ""));
+            return new UnMarkCommand(unMarkNum - 1);
+            // Fallthrough
+        case TODO:
+            String tDes = command.replace("todo", "");
+            return new AddCommand(new Todo(tDes));
+            // Fallthrough
```

Comment:
This works but having a `// Fallthrough` comment feels redundant since we are already returning here, would removing it be more appropriate?